src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -96,26 +96,39 @@ private static void updateKnownPaths(@SuppressWarnings("SameParameterValue") @No
    * Adds the current path and other known paths to the combo, most recently used first.
    */
   public static void addKnownSDKPathsToCombo(@NotNull JComboBox combo) {
-    final Set<String> pathsToShow = new LinkedHashSet<>();
-
+    // First, get the current path from the combo box on the EDT.
     final String currentPath = combo.getEditor().getItem().toString().trim();
+    final Set<String> pathsToShow = new LinkedHashSet<>();
     if (!currentPath.isEmpty()) {
       pathsToShow.add(currentPath);
     }
 
-    final String[] knownPaths = getKnownFlutterSdkPaths();
-    for (String path : knownPaths) {
-      if (FlutterSdk.forPath(path) != null) {
-        pathsToShow.add(FileUtil.toSystemDependentName(path));
+    // Now, run the slow operation (finding valid SDK paths) on a background thread.
+    ApplicationManager.getApplication().executeOnPooledThread(() -> {
```

Comment:
Neat! I'll have to read up on how this thread pool is managed and what kinds of operations are best run on it. Maybe this is the approach we should take elsewhere?

Reply:
It is part of it, I was reading up on `coalesceBy`, see the message in https://github.com/flutter/flutter-intellij/issues/8242, which I think will be the right approach.